subsys/bluetooth/controller/power_clock_driver.c

```diff
@@ -10,20 +10,32 @@
 #include <kernel_includes.h>
 #include <clock_control.h>
 #include <ble_controller_soc.h>
+#include "multithreading_lock.h"
 
 static int hf_clock_start(struct device *dev, clock_control_subsys_t sub_system)
 {
 	ARG_UNUSED(dev);
 
-	if (ble_controller_hf_clock_request(NULL) != 0) {
+	int32_t errcode = MULTITHREADING_LOCK_ACQUIRE();
+	if (errcode == 0) {
+		errcode = ble_controller_hf_clock_request(NULL);
+		MULTITHREADING_LOCK_RELEASE();
+	}
+	if (errcode != 0) {
 		return -EFAULT;
```

Comment:
This is overriding the potential error code from `MULTITHREADING_LOCK_AQUIRE()` is that intentional?

Reply:
Yes. Even in the earlier case, this driver was returning -EFAULT when API call was unsuccessful. These changes keep the same behaviour. Probably that is how the driver interface works. Returning some other error code might break the user of this driver.